Mathlib/AlgebraicTopology/SimplicialSet/Basic.lean

```diff
@@ -444,4 +444,174 @@ noncomputable def standardSimplex : SimplexCategory ⥤ SSet.Augmented.{u} where
 
 end Augmented
 
+section Spine
+
+variable (X : SSet.{u})
+
+/-- A path in a simplicial set `X` of length `n` is a directed path of `n` edges.-/
+@[ext]
+structure Path (n : ℕ) where
+  /-- A path includes the data of `n+1` 0-simplices in `X`.-/
+  vertex (i : Fin (n + 1)) : X _[0]
+  /-- A path includes the data of `n` 1-simplices in `X`.-/
+  arrow (i : Fin n) : X _[1]
+  /-- The sources of the 1-simplices in a path are identified with appropriate 0-simplices.-/
+  arrow_src (i : Fin n) : X.δ 1 (arrow i) = vertex i.castSucc
+  /-- The targets of the 1-simplices in a path are identified with appropriate 0-simplices.-/
+  arrow_tgt (i : Fin n) : X.δ 0 (arrow i) = vertex i.succ
+
+/-- For `j ≤ k ≤ n`, a path of length `n` restricts to a path of length `k-j`, namely the subpath
+spanned by the vertices `j ≤ i ≤ k` and edges `j ≤ i < k`. -/
+def Path.interval {n : ℕ} (f : Path X n) (j k : ℕ) (hjk : j ≤ k) (hkn : k ≤ n) :
```

Comment:
Using the operator `-` on `ℕ` or `Fin n` creates a lot of annoyance. I suggest parametrizing the tuples `(j ≤ k)` which appear in several places as `(j, j + l)`:
```lean
def Path.interval {n : ℕ} (f : Path X n) (j l : ℕ) (hjl : j + l ≤ n) :
    Path X l where
  vertex i := f.vertex ⟨j + i, by omega⟩
  arrow i := f.arrow ⟨j + i, by omega⟩
  arrow_src i := f.arrow_src ⟨j + i, by omega⟩
  arrow_tgt i := f.arrow_tgt ⟨j + i, by omega⟩
```